### PR TITLE
limit background loader jobs on sync on mobile

### PR DESCRIPTION
**Pull request by mmaxim**
* Limit background loading on mobile to max number of convs, and only the first page. If in the foreground, we really limit the loading to just the top convs.
* Sort the convs based on time, and also deprioritize big team convs.

### CODE REVIEW

File: go/chat/sync.go [hunk 1]
```diff
@@ -458,11 +465,34 @@ func (s *Syncer) sync(ctx context.Context, cli chat1.RemoteInterface, uid gregor
...
-			maxConvs = 3
+		pageBack := 3
+		num := 50
+		netState := s.G().MobileNetState.State()
+		state := s.G().MobileAppState.State()
+		if s.G().IsMobileAppType() {
+			maxConvs = s.maxConvLoads
+			num = 30
+			pageBack = 0
+			if netState.IsLimited() || state == keybase1.MobileAppState_FOREGROUND {
```
Comment (joshblum) [inline go/chat/sync.go]: how do we know we just came into the foreground here?

File: go/chat/sync.go [hunk 2]
```diff
@@ -491,14 +521,18 @@ func (s *Syncer) sync(ctx context.Context, cli chat1.RemoteInterface, uid gregor
 				}
 				queuedConvs++
 			} else {
-				// If we are a limited data connection only queue selected
-				// convs up to maxConvs
-				if state.IsLimited() && (queuedConvs >= maxConvs || !s.shouldUnboxSyncConv(conv)) {
+				// Only do this for chat convs
+				if conv.GetTopicType() != chat1.TopicType_CHAT {
```
Comment (joshblum) [inline go/chat/sync.go]: why not move this into `shouldUnboxSyncConv`?

Comment (mmaxim) [reply go/chat/sync.go]: Need KBFS convs...

File: go/chat/sync.go (hunk 1 above)
Comment (mmaxim) [reply go/chat/sync.go]: Just the foreground check is good enough

File: go/chat/sync.go (hunk 2 above)
Comment (mmaxim) [reply go/chat/sync.go]: Maybe we can ignore them on mobile.

**Pull request merged**